### DISCUSSION TITLE
File (base64) in JSON Body

### DISCUSSION THREAD

**Discussion by rons4** [Waboodoo/HTTP-Shortcuts#408] [Q&A] [answered]
Hi,

I recently came across HTTP Shortcuts and was really impressed.

Since it is possible to upload photos, I wonder if there is a possibility to insert the photo file as base64 somewhere in a JSON payload of an HTTP Post request.

So, something like this:

`
{
"Name": "My Image",
"Content-Type": "image/jpg",
"Content-Transfer-Encoding": "base64",
"Content": "--the-content-of-the-photo-file-as-base64-string--"
}
`

It is possible for example in conjunction with Tasker. Is it possible within HTTP Shortcut directly?

Thanks

Ron

**Comment by Waboodoo** [comment, marked as answer]
Hi,
Currently, this is not possible with the app, due to its limited capabilities of interacting with files. I have a vague vision of one day introducing a feature that allows you to read arbitrary files as part of the [Scripting](https://http-shortcuts.rmy.ch/scripting) feature, which would enable something like this, but it will be a while before that becomes a reality.
I'm not aware of a way to do this with Tasker. It would need to read the file and send its contents as a variable, ideally already base64-encoded. It could then be received in a variable and used by HTTP Shortcuts. But this would only work with very small files, as it would quickly run into the size limit that Android enforces for inter-app-communication.

**Comment by rons4** [comment]
Thanks a lot. In the meantime I figured out how to use form-data instead. I wanted to use this for sending an alert and it works quite well.
I even created a short video: https://youtu.be/h6dKCPlA2h0

I also shared this on LinkedIn and X in case you want to share or like ;-)

LinkedIn:
https://www.linkedin.com/posts/czachara_signl4-alerting-send-a-photo-as-part-of-activity-7161759331292585984-FBLN?utm_source=share&utm_medium=member_desktop

X:
https://x.com/roncza/status/1755993888538435629?s=20

By the way, for sending images using Tasker you can find more information here: https://www.reddit.com/r/tasker/comments/f3qbvc/how_to_send_images_via_telegram_messenger_or/
However, this is a bit more complicated.

**Reply by Waboodoo** [reply]
Glad to hear you found a solution. Thanks for the video, it makes it all look so simple :)
